### PR TITLE
Bump Kotlin Coroutines dependency to 1.4.2

### PR DESCRIPTION
**Pull request by twyatt**
`SharedFlow` now works on `-native-mt` variant of Coroutines, so `BroadcastChannel` was dropped in favor of `SharedFlow`.

### CODE REVIEW

**Review by sdonn3 — APPROVED** [review]

File: core/src/macosX64Main/kotlin/Flow.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.juul.kable
+
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.runBlocking
+
+internal fun <T> MutableSharedFlow<T>.sendBlocking(value: T) {
```
Comment (cedrickcooke) [inline core/src/macosX64Main/kotlin/Flow.kt:6]: Nitpick:
```suggestion
internal fun <T> MutableSharedFlow<T>.emitBlocking(value: T) {
```

**Review by cedrickcooke — APPROVED** [review]

File: core/src/macosX64Main/kotlin/Flow.kt (hunk 1 above)
Comment (twyatt) [reply core/src/macosX64Main/kotlin/Flow.kt:6]: Oh, good catch. Will change. Thanks!

File: core/src/macosX64Main/kotlin/CentralManagerDelegate.kt [hunk 2]
```diff
@@ -24,11 +22,8 @@ import kotlin.native.concurrent.freeze
 // https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerdelegate
 internal class CentralManagerDelegate : NSObject(), CBCentralManagerDelegateProtocol {
 
-    // todo: MutableSharedFlow
-    private val _onDisconnected = BroadcastChannel<NSUUID>(1)
-    internal val onDisconnected: Flow<NSUUID>
-        // todo: Drop `get()` when `_onDisconnected` is a MutableSharedFlow.
-        get() = _onDisconnected.openSubscription().consumeAsFlow()
+    private val _onDisconnected = MutableSharedFlow<NSUUID>()
+    internal val onDisconnected = _onDisconnected.asSharedFlow()
```
Comment (Phoenix7351) [inline core/src/macosX64Main/kotlin/CentralManagerDelegate.kt:26]: Just curious, is a MutableSharedFlow not _already_ a shared flow?

Comment (twyatt) [reply core/src/macosX64Main/kotlin/CentralManagerDelegate.kt:26]: `MutableSharedFlow` is a mutable version of `SharedFlow`.

Using `asSharedFlow` exposes the underlying (`_onDisconnected`) **mutable** version of the flow, as a read-only `SharedFlow`.

File: core/src/macosX64Main/kotlin/Peripheral.kt [hunk 3]
```diff
@@ -130,9 +131,12 @@ public class ApplePeripheral internal constructor(
                 _connection.value = it
             }
 
-            connection.characteristicChanges
+            connection
+                .delegate
+                .characteristicChanges
+                .takeWhile { it !== Closed }
```
Comment (Phoenix7351) [inline core/src/macosX64Main/kotlin/Peripheral.kt:137]: I don't know what type Closed is, but did you intentionally use the referential operator? (!==)

Comment (twyatt) [reply core/src/macosX64Main/kotlin/Peripheral.kt:137]: `Closed` is a "singleton" `object`, so the `!==` is a small optimization for that comparison. `!=` would've worked (slightly less efficient though).

https://github.com/JuulLabs/kable/blob/09619e55d9cdd8e62edad873c1d916bdd5aa1fab/core/src/macosX64Main/kotlin/PeripheralDelegate.kt#L79-L93

File: core/src/macosX64Main/kotlin/CentralManagerDelegate.kt (hunk 2 above)
Comment (Phoenix7351) [reply core/src/macosX64Main/kotlin/CentralManagerDelegate.kt:26]: Gotcha. So privately you want it mutable and immutable internally.

File: core/src/macosX64Main/kotlin/Peripheral.kt (hunk 3 above)
Comment (Phoenix7351) [reply core/src/macosX64Main/kotlin/Peripheral.kt:137]: Got it. Thanks.

File: core/src/macosX64Main/kotlin/CentralManagerDelegate.kt (hunk 2 above)
Comment (twyatt) [reply core/src/macosX64Main/kotlin/CentralManagerDelegate.kt:26]: Ya, basically isolating that emissions on the `Flow` can only come from within this class (and that downstream consumers can never also push items onto the stream — can only read from the stream).

Basically, just good practice to help keep the overall system simpler and less error prone.

**Pull request merged**